internal/handler/msg_hello.go

```diff
@@ -36,21 +40,112 @@ func (h *Handler) MsgHello(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, e
 		return nil, lazyerrors.Error(err)
 	}
 
+	saslSupportedMechs, err := common.GetOptionalParam(doc, "saslSupportedMechs", "")
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	var saslSupportedMechsResp *types.Array
+
+	if saslSupportedMechs != "" {
```

Comment:
This is just a suggestion to reduce nested `if`s, what do you think of handling empty `saslSupportedMechs` case first and returning it before we handle non empty saslSupportedMechs? :thinking: 

```
	var reply wire.OpMsg
	resp := must.NotFail(types.NewDocument(
		"isWritablePrimary", true,
		"maxBsonObjectSize", int32(types.MaxDocumentLen),
		"maxMessageSizeBytes", int32(wire.MaxMsgLen),
		"maxWriteBatchSize", int32(100000),
		"localTime", time.Now(),
		"connectionId", int32(42),
		"minWireVersion", common.MinWireVersion,
		"maxWireVersion", common.MaxWireVersion,
		"readOnly", false,
	))

	if saslSupportedMechs == "" {
		resp.Set("ok", float64(1))

		must.NoError(reply.SetSections(wire.MakeOpMsgSection(resp)))

		return &reply, nil
	}

	db, username, ok := strings.Cut(saslSupportedMechs, ".")
	// path with non empty saslSupportedMechs
```